// src/components/common/layouts/LayoutAuthorization/textsLang.ts
export const textsLang = {
  ru: {
    authButton: 'ВХОД',
    forgotPassword: 'ЗАБЫЛИ ПАРОЛЬ?',
    authorization: {
      msgDefault: 'С возвращением!',
      submit: 'Войти'
    },
    registration: {
      msgDefault: 'Присоединяйтесь!',
      submit: 'Зарегистрироваться'
    },
    reset: {
      msgDefault: 'Введите email для сброса пароля',
      submit: 'Отправить'
    },
    msgErrorInputEmail: 'Проверьте введенный Email',
    msgErrorInputPassword: 'Пароль должен быть от 6 до 32 символов',

    resetOk: 'Мы отправили письмо на ваш Email. Перейдите по ссылке в письме для установки нового пароля.',

    regOk: 'Мы отправили письмо на ваш Email. Перейдите по ссылке в письме для завершения регистрации.',
    msgRegUserExists: 'Пользователь с таким адресом электронной почты уже существует',
    msgRegMailSendError: 'Мы не смогли отправить письмо. Такого Email не существует',

    msgAuthWrongPassword: 'Неправильный пароль',
    msgAuthNoUser: 'Пользователя с таким Email не существует',

    msgDbError: 'Ошибка сервера',
    msgError: 'Неизвестная ошибка'
  },
  en: {
    authButton: 'LOG IN',
    forgotPassword: 'FORGOT PASSWORD?',
    authorization: {
      msgDefault: 'Welcome back!',
      submit: 'Log in'
    },
    registration: {
      msgDefault: 'Join now!',
      submit: 'Sign up'
    },
    reset: {
      msgDefault: 'Enter email for reset password',
      submit: 'Submit'
    },
    msgErrorInputEmail: 'Check entered Email',
    msgErrorInputPassword: 'Password must be between 6 and 32 characters',

    resetOk: 'We have sent an mail to your Email. Follow the link in the letter to set a new password.',

    regOk: 'We have sent an mail to your Email. Follow the link in the letter to complete the registration',
    msgRegUserExists: 'User with this email already exists',
    msgRegMailSendError: 'We were unable to send a letter. This Email does not exist',

    msgAuthWrongPassword: 'Wrong Password',
    msgAuthNoUser: 'User with such Email does not exist',

    msgDbError: 'Server error',
    msgError: 'Unknown error'
  }
}
